pycbc/distributions/joint.py

```diff
@@ -242,6 +242,39 @@ def contains(self, params):
             result = result.item()
         return result
 
+    def __contains__(self, params):
```

Comment:
@WuShichao Thank you for putting this together. I think it is nearly ready to go. Given that as you've shown you can do array input with the 'in' operator (this appears to be a limitation of the __contains__ operator itself !, I think perhaps it would be good to rename this method from __contains__ to something else. 

I think the current 'contains' method is only used in a few places. I propose we swap this out as that name is confusing. Perhaps it can be renamed to 'within_constraints' and your __constraints__ should be renamed to simply 'constrains'. That would mean renaming any existing uses 'of constrains' to point to 'within_constraints' instead.